cypress/helpers/customTypes.js

```diff
@@ -33,30 +32,49 @@ export function renameCustomType(id, actualName, newName) {
     `/ ${newName} - Edited`
   );
   cy.readFile(TYPES_FILE).should("contains", `${newName} - Edited`);
+  cy.readFile(CUSTOM_TYPE_MODEL(id)).then((model) => {
+    expect(JSON.stringify(model)).to.contain(newName);
+  });
 }
 
-export function addFieldToCustomType(
-  id,
-  fieldType,
-  fieldName,
-  fieldId,
-  isFirstField /* boolean | undefined */
-) {
-  const addFieldButtonSelector = isFirstField
-    ? "empty-zone-add-new-field"
-    : "add-new-field";
-
-  cy.get(`[data-cy="${addFieldButtonSelector}"]`).first().click();
+export function addFieldToCustomType(fieldType, fieldName, fieldId) {
+  cy.get(`[data-cy="add-Static-field"]`).first().click();
   cy.get(`[data-cy='${fieldType}']`).click();
-  cy.get("[data-cy=new-field-name-input]").clear().type(fieldName);
+
+  cy.get("[data-cy=new-field-name-input]").clear();
+  // waiting for the field to re-render
+  cy.wait(500);
+  cy.get("[data-cy=new-field-name-input]").type(fieldName);
 
   // API Id modification for UID field is disabled
-  if (fieldType != "UID")
-    cy.get("[data-cy=new-field-id-input]").clear().type(fieldId);
+  if (fieldType != "UID") {
+    cy.get("[data-cy=new-field-id-input]").clear();
+    // waiting for the field to re-render
+    cy.wait(500);
+    cy.get("[data-cy=new-field-id-input]").type(fieldId);
+  }
 
   cy.get("[data-cy=new-field-form]").submit();
   cy.get("[data-cy=ct-static-zone]").within(() => {
     cy.contains(fieldName).should("be.visible");
     cy.contains(`data.${fieldId}`).should("be.visible");
   });
 }
+
+export function addSlicesToCustomType(sliceIds /* string[] */) {
+  cy.get("[data-cy=update-slices]").click();
+
+  sliceIds.forEach((sliceId) => {
+    // forcing this because the input itself is invisible and an svg is displayed
+    cy.get(`[data-cy=check-${sliceId}]`).click({ force: true });
+  });
+
+  cy.get("[data-cy=update-slices-modal]").submit();
+}
+
+export function saveCustomTypeModifications() {
+  cy.get("[data-cy=builder-save-button]").should("not.be.disabled");
+  cy.get("[data-cy=builder-save-button]").click();
+  cy.get("[data-cy=builder-save-button-spinner]").should("be.visible");
```

Comment:
Should we split actions and checks? Looking at the function name it's not easy to know what it does

Reply:
We could definitely split actions and checks but I'd expect the button not to be clicked if it isn't clickable here for example.